qa/pull-tester/rpc-tests.py

```diff
@@ -179,49 +118,125 @@
+    parser.add_argument('--extended', action='store_true', help='run the extended test suite in addition to the basic tests')
+    parser.add_argument('--help', '-h', '-?', action='store_true', help='print help text and exit')
+    parser.add_argument('--nozmq', action='store_true', help='do not run the zmq tests')
+    parser.add_argument('--parallel', type=int, default=4, help='how many test scripts to run in parallel. Default=4.')
+    parser.add_argument('--win', action='store_true', help='signal that this is running in a Windows environment and that we should run the tests')
+    (args, unknown_args) = parser.parse_known_args()
+
+    # Create a set to store arguments and create the passon string
+    tests = set()
+    passon_args = []
+
+    for arg in unknown_args:
+        if arg[:2] == "--":
+            passon_args.append(arg)
+        else:
+            tests.add(arg)
+
+    # Read config generated by configure.
+    config = configparser.ConfigParser()
+    config.read_file(open(os.path.dirname(__file__) + "/tests_config.ini"))
+
+    enable_wallet = config["components"]["ENABLE_WALLET"] == "True"
+    enable_utils = config["components"]["ENABLE_UTILS"] == "True"
+    enable_bitcoind = config["components"]["ENABLE_BITCOIND"] == "True"
+    enable_zmq = config["components"]["ENABLE_ZMQ"] == "True" and not args.nozmq
+
+    if config["environment"]["EXEEXT"] == ".exe" and not args.win:
+        # https://github.com/bitcoin/bitcoin/commit/d52802551752140cf41f0d9a225a43e84404d3e9
+        # https://github.com/bitcoin/bitcoin/pull/5677#issuecomment-136646964
+        print("Win tests currently disabled by default.  Use -win option to enable")
+        sys.exit(0)
+
+    if not (enable_wallet and enable_utils and enable_bitcoind):
+        print("No rpc tests to run. Wallet, utils, and bitcoind must all be enabled")
+        print("Rerun `configure` with -enable-wallet, -with-utils and -with-daemon and rerun make")
+        sys.exit(0)
+
+    # python3-zmq may not be installed. Handle this gracefully and with some helpful info
+    if enable_zmq:
+        try:
+            import zmq
+        except ImportError:
+            print("ERROR: \"import zmq\" failed. Use -nozmq to run without the ZMQ tests."
+                  "To run zmq tests, see dependency info in /qa/README.md.")
+            raise
+
     # Build list of tests
     test_scripts = BASE_SCRIPTS
-    if ENABLE_ZMQ:
+    if enable_zmq:
         test_scripts += ZMQ_SCRIPTS
     if args.extended:
         test_scripts += EXTENDED_SCRIPTS
 
     if len(tests) == 0:
         test_list = test_scripts
     else:
-        test_list = []
-        for t in test_scripts:
-            if t in tests or re.sub(".py$", "", t) in tests:
-                test_list.append(t)
+        test_list = [t for t in test_scripts if 
+                     (t in ALL_SCRIPTS or re.sub(".py$", "", t) in ALL_SCRIPTS)]
```

Comment:
You're completely right. I put the ALL_SCRIPTS in the wrong place.

Should be fixed now.